orchagent/pfcwdorch.cpp

```diff
@@ -818,27 +890,89 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::doTask(swss::NotificationConsumer
 
             entry->second.handler->commitCounters();
             entry->second.handler = nullptr;
+            // Remove storm status in APPL_DB for warm-reboot purpose
+            string key = APP_PFC_WD_TABLE_NAME ":" + entry->second.portAlias;
+            m_applDbRedisClient.hdel(key, to_string(entry->second.index));
         }
     }
     else
     {
         SWSS_LOG_ERROR("Received unknown event from plugin, %s", event.c_str());
+        return false;
     }
+
+    return true;
 }
 
 template <typename DropHandler, typename ForwardHandler>
-void PfcWdSwOrch<DropHandler, ForwardHandler>::doTask(SelectableTimer &timer)
+bool PfcWdSwOrch<DropHandler, ForwardHandler>::bake()
 {
-    SWSS_LOG_ENTER();
+    // clean all *_last fields in COUNTERS_TABLE
+    // to allow warm-reboot pfc detect & restore state machine to enter the same init state as cold-reboot
+    RedisClient redisClient(PfcWdOrch<DropHandler, ForwardHandler>::getCountersDb().get());
 
-    for (auto& handlerPair : m_entryMap)
+    vector<string> cKeys;
+    PfcWdSwOrch<DropHandler, ForwardHandler>::getCountersTable()->getKeys(cKeys);
+    for (const auto &key : cKeys)
     {
-        if (handlerPair.second.handler != nullptr)
+        vector<FieldValueTuple> fvTuples;
+        PfcWdSwOrch<DropHandler, ForwardHandler>::getCountersTable()->get(key, fvTuples);
+        for (const auto &fv : fvTuples)
         {
-            handlerPair.second.handler->commitCounters(true);
+            if (fvField(fv).find("_last") != string::npos)
+            {
+                redisClient.hdel(COUNTERS_TABLE ":" + key, fvField(fv));
```

Comment:
>hdel [](start = 28, length = 4)

optimize to 1 redis call. ref: [https://redis.io/commands/HDEL](https://redis.io/commands/HDEL)

Reply:
This is an infrastructure lack. Added in https://github.com/Azure/sonic-swss-common/pull/267